src/precice/impl/ParticipantImpl.cpp

```diff
@@ -1618,4 +1643,67 @@ ParticipantImpl::MappedSamples ParticipantImpl::mappedSamples() const
   return res;
 }
 
+// Reinitialization
+
+int ParticipantImpl::getTotalMeshChanges() const
+{
+  PRECICE_TRACE();
+  PRECICE_ASSERT(_allowsRemeshing);
+  Event e("intra-handshake", profiling::Synchronize);
+  int   localMeshesChanges = _meshLock.countUnlocked();
+  PRECICE_DEBUG("Local Mesh Changes: {}", localMeshesChanges);
+
+  int totalMeshesChanges = 0;
+  utils::IntraComm::allreduceSum(localMeshesChanges, totalMeshesChanges);
+  PRECICE_DEBUG("Total Mesh Changes: {}", totalMeshesChanges);
+  return totalMeshesChanges;
+}
+
+bool ParticipantImpl::reinitHandshake(bool requestReinit) const
```

Comment:
We currently still reinit all m2ns, right? Not only those, which participants share a reset mesh?

Reply:
Yes, this is currently restricted in the configuration though.